### PR TITLE
fix(BModal): Fix new modals stacking behind active modals

### PR DESCRIPTION
**Pull request by mclmax**
# Describe the PR

This PR introduces a small change to how the z-indices of active modals are calculated.

Currently there are cases ([issue 1516](https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1516)) where a newly opened modal will open behind an active modal.

The change to behaviour here is to calculate the z-index of a modal based on it's position in the stack of active modals. All modals have a z-index lower than or equal to 1055. The newer (high up on the stack) the modal is the close to 1055. The latest having a z-index of 1055. Inactive modals have a z-index of 1055 by default, the highest value. Meaning when they open they are on top by default until the z-index is adjusted.

There is a slightly different approach I tried which was just to calculate the z-index as `1055 + indexInStack` however, this meant that for a small time window the new modal would be behind any active modal until the new z-index was calculated. This meant that the modal wouldn't fade in smoothly and instead would pop in quite abruptly. 

## Small replication

A small replication or video walkthrough can help demonstrate the changes made. This is optional, but can help observe the intended changes. A mentioned issue that contains a replication also works.

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: packages/bootstrap-vue-next/src/composables/useModalManager.ts [hunk 1]
```diff
@@ -62,4 +72,9 @@ export default (modalOpen: Ref<boolean>): void => {
     },
     {immediate: true}
   )
+
+  return {
+    activePosition: computed(() => stack.value.findIndex((el) => el.exposed?.id === toValue(id))),
+    activeModalCount: computed(() => stack.value.length),
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/composables/useModalManager.ts:78]:
```suggestion
    activeModalCount: toRef(() => stack.value.length),
```

File: packages/bootstrap-vue-next/src/components/BModal.vue [hunk 2]
```diff
@@ -506,7 +515,17 @@ const onAfterLeave = () => {
...
-useModalManager(isActive)
+const {activePosition, activeModalCount} = useModalManager(isActive, computedId)
+const defaultModalDialogZIndex = 1056
+const computedZIndex = computed<CSSProperties>(() => ({
+  // Make sure that newly opened modals have a higher z-index than currently active ones.
+  // All active modals have a z-index of ('defaultZIndex' - 'stackSize' - 'positionInStack').
+  //
+  // This means inactive modals will already be higher than active ones when opened.
+  'z-index': isActive.value
+    ? defaultModalDialogZIndex - (activeModalCount.value - activePosition.value)
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BModal.vue:526]: What affect does having `(activeModalCount.value` in here have

**Review by VividLemon — CHANGES_REQUESTED** [review]

File: packages/bootstrap-vue-next/src/components/BModal.vue (hunk 2 above)
Comment (mclmax) [reply packages/bootstrap-vue-next/src/components/BModal.vue:526]: it's used to flip the "priority" of the modals on the stack. so if we have [Modal1, Modal2, Modal3] (0, 1, 2), we want Modal3 to have the highest z-index, Modal2 second highest etc. without the count we would have [1056, 1055, 1054] which isn't what we want. subtracting stack size first gives: [1053, 1054, 1055].

Comment (mclmax) [reply packages/bootstrap-vue-next/src/components/BModal.vue:526]: there is the risk that things could get slightly weird if folks are opening hundreds of modals i guess

File: packages/bootstrap-vue-next/src/composables/useModalManager.ts (hunk 1 above)
Comment (mclmax) [reply packages/bootstrap-vue-next/src/composables/useModalManager.ts]: made the change

**Pull request merged**